Cabal/src/Distribution/Simple/Test.hs

```diff
@@ -58,26 +69,32 @@ test
   -> TestFlags
   -- ^ flags sent to test
   -> IO ()
-test args pkg_descr lbi flags = do
+test args pkg_descr lbi0 flags = do
   let verbosity = fromFlag $ testVerbosity flags
       machineTemplate = fromFlag $ testMachineLog flags
       distPref = fromFlag $ testDistPref flags
       testLogDir = distPref </> "test"
       testNames = args
       pkgTests = PD.testSuites pkg_descr
       enabledTests = LBI.enabledTestLBIs pkg_descr lbi
+      internalPkgDB = internalPackageDBPath lbi distPref
+      lbi = lbi0{withPackageDB = withPackageDB lbi0 ++ [SpecificPackageDB internalPkgDB]}
 
       doTest
-        :: ( (PD.TestSuite, LBI.ComponentLocalBuildInfo)
+        :: [FilePath]
+        -- \^ The paths to the library components to include in the coverage report
+        -> [ModuleName]
+        -- \^ The modules to include in the coverage report
+        -> ( (PD.TestSuite, LBI.ComponentLocalBuildInfo)
            , Maybe TestSuiteLog
            )
         -> IO TestSuiteLog
-      doTest ((suite, clbi), _) =
+      doTest testCoverageDistPrefs testCoverageLibsModules ((suite, clbi), _) =
         case PD.testInterface suite of
           PD.TestSuiteExeV10 _ _ ->
-            ExeV10.runTest pkg_descr lbi clbi flags suite
+            ExeV10.runTest pkg_descr lbi clbi testCoverageDistPrefs testCoverageLibsModules flags suite
```

Comment:
perhaps testCoverageDistPrefs should be computed inside `ExeV10.runTest` rather than passed in?

Reply:
Or perhaps we should make a datatype to pass around this folders and modules to generate reports for.

Reply:
Great idea, I'll make the datatype.